.gitlab-ci.yml

```
@@ -159,6 +169,19 @@ default:
     # this job runs only on nightly pipeline with the mentioned variable, against `master` branch
     - if: $CI_COMMIT_REF_NAME == "master" && $CI_PIPELINE_SOURCE == "schedule" && $PIPELINE == "nightly"
 
+.build-rustdoc-rules:              &build-rustdoc-rules
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "pipeline"
+      when: never
+    - if: $CI_PIPELINE_SOURCE == "web" && $RM_RUSTDOC_REF
+      when: never
+    - if: $CI_PIPELINE_SOURCE == "schedule" || $CI_PIPELINE_SOURCE == "push" ||
+        ($CI_PIPELINE_SOURCE == "web" && $BUILD_RUSTDOC_REF)
```

```suggestion
    - if: $CI_PIPELINE_SOURCE == "schedule" && $BUILD_RUSTDOC_REF
```
explained above, plus `$CI_PIPELINE_SOURCE == "push"` does not work in the mirrored setup.